src/newgrf_industries.cpp

```diff
@@ -575,13 +597,28 @@ void IndustryProductionCallback(Industry *ind, int reason)
 		if (tgroup == NULL || tgroup->type != SGT_INDUSTRY_PRODUCTION) break;
 		const IndustryProductionSpriteGroup *group = (const IndustryProductionSpriteGroup *)tgroup;
 
-		bool deref = (group->version == 1);
+		bool deref = (group->version >= 1);
 
-		for (uint i = 0; i < 3; i++) {
-			ind->incoming_cargo_waiting[i] = Clamp(ind->incoming_cargo_waiting[i] - DerefIndProd(group->subtract_input[i], deref) * multiplier, 0, 0xFFFF);
-		}
-		for (uint i = 0; i < 2; i++) {
-			ind->produced_cargo_waiting[i] = Clamp(ind->produced_cargo_waiting[i] + max(DerefIndProd(group->add_output[i], deref), 0) * multiplier, 0, 0xFFFF);
+		if (group->version < 2) {
```

Comment:
The comments here are misleading.
This "if" is about whether group->cargo_input/output is valid and shall be used.
Even though the number of cargos are known for version < 2, I would still use num_input/output there, since they are set in newgrf.cpp.